### ISSUE TITLE
[question] Add custom generated classes to docker image structure under classes

### ISSUE THREAD

**Issue by eshepelyuk** [bmuschko/gradle-docker-plugin#855]
I'm having custom task called `myGenTask` that generates some byte code to separate directory under `${project.buildDir}`, let's say the folder is `build/my-gen-clz`. And this directory is declated as task output.

I'm using `com.bmuschko.docker-java-application` plugin for creating docker image structure.

### Current Behavior

Those generated classes are not included in resulting docker files after running `dockerSyncBuildContext`.

Only main sourceSets's classesDir can be automatically added to `classes` folder of `docker` image files.

https://github.com/bmuschko/gradle-docker-plugin/blob/master/src/main/groovy/com/bmuschko/gradle/docker/utils/ConventionPluginHelper.groovy#L42

### Expected Behavior

I'm able to workaround the problem by adding this to `build.gradle`

```
dockerSyncBuildContext {
  into("classes") {
    from myGenTask.outputs
  }
}
```

But this should be somehow more straightforward. Unfortunately I can't find a way to modify `classesDir` property on `SourceSetOutput`.

### Your Environment

Gradle 5.6.2
Plugin 5.0.0
JDK 1.8

**Comment by cdancy** [comment]
@eshepelyuk is there any particular reason you can't have your "bytecode generator" point at the `classes` folder? We always try to create a reasonable standard to follow so that folks have a simple and clear path to follow. Your workaround looks sane enough (and is what I would have done) and keep inline with the documentation [HERE](https://github.com/bmuschko/gradle-docker-plugin/blob/93c0b3d63c371d8ebc11105830b00a490d260eb6/src/docs/asciidoc/23-tasks.adoc). 

We could _potentially_ have an additional "generated" folder as I've seen lots of projects use that pattern.

**Comment by eshepelyuk** [comment]
I've been able to come up with the ultimate (and I believe 100% proper) solution for this issue.
I've configured that task outputs to be considered as a part of `main` sourceSet output by

```
project.sourceSets.main.output.classesDirs.from(myGenTask.outputs)
```

So now 
- generated classes are included in generated project jar without additional setup
- generated classes are picked up by this plugin

Although it could be good if such cases could be configured in more straightforward way.
Closing the issue.
